docs/component_conventions.md

```diff
@@ -19,14 +19,45 @@ The namespace indicates where a component lives. A single page on GOV.UK could r
 
 | Type | Location | Example | Description |
 | -- | -- | -- | -- |
-| Template | `app/views/components` | `_my-comp.html.erb` | The template logic and markup. The template defines the component’s API. Filename must begin with an underscore. |
+| Template | `app/views/components` | `_my-comp.html.erb` | [Template logic and markup](#template) |
 | Documentation | `app/views/components/docs` | `my-comp.yml` | [Describes the component](#write-documentation) |
 | Styles | `app/assets/stylesheets/components` | `_my-comp.scss` | [Component styles](#styles) |
+| Print styles | `app/assets/stylesheets/components/print` | `_my-comp.scss` | [Component styles](#styles) |
 | Images | `app/assets/images/govuk_publishing_components` | `my-comp.png` | [Images](#images) |
 | Scripts | `app/assets/javascripts/components` | `my-comp.js` | [Javascript enhancements](#javascript) |
-| Tests | `test/components` | `my_comp_test.rb` | Unit tests |
+| Tests | `test/components` | `my_comp_test.rb` | [Unit tests](#tests) |
+| Javascript tests | `spec/components` | `my_comp_spec.rb` | [Unit tests](#tests) |
 | Helpers | `lib/govuk_publishing_components/presenters` | `my_comp_helper.rb` | [Helpers](#helpers) |
 
+## Template
+
+The template logic and markup. The template defines the component’s API. Filename must begin with an underscore.
+
+If complex logic is required this should be handed using a [helper](#helpers).
+
+Example:
+
+```ruby
+<%
+  options ||= []
+  id ||= false
+  helper = GovukPublishingComponents::Presenters::MyComponentHelper.new(options)
+%>
+
+<% if options.any? %>
+  <div class="govuk-something" id="<%= id %>">
+    <h2>An example component</h2>
+    <%= helper.someContent %>
+  </div>
+<% endif %>
+```
+
+If a component includes a heading, there should be an option to control the heading level (see the [heading component](https://govuk-publishing-components.herokuapp.com/component-guide/heading/specific_heading_level) for example).
+
+Components can use other components within their template, if required.
```

Comment:
Can we link to an example? It might also be worth mentioning splitting complex components into partials and including them in the component, as we have examples of those.